### PR TITLE
check whether notif level is undefined, because `0` is falsey 

### PR DESCRIPTION
**Pull request by t3chguy**
so if you set `@room` notif=0, then js-sdk would think its actually 50 :(

Signed-off-by: Michael Telatynski <7t3chguy@gmail.com>

### CODE REVIEW

File: src/models/room-state.js [hunk 1]
```diff
@@ -455,7 +455,7 @@ RoomState.prototype.mayTriggerNotifOfType = function(notifLevelKey, userId) {
         powerLevelsEvent &&
         powerLevelsEvent.getContent() &&
         powerLevelsEvent.getContent().notifications &&
-        powerLevelsEvent.getContent().notifications[notifLevelKey]
+        powerLevelsEvent.getContent().notifications[notifLevelKey] !== undefined
```
Comment (turt2live) [inline src/models/room-state.js]: and if it's `null`, `false`, `NaN`, or `[]`? Might be best to change the check to be something like this:

```
!isNaN(powerLevelsEvent.getContent().notifications[notifLevelKey]) 
&& typeof(powerLevelsEvent.getContent().notifications[notifLevelKey]) === 'number'
```

**Comment by t3chguy** [comment]
I kept it consistent with the other checks but did a not NaN previously, will make another pr to change them all after this one. I guess here we're trusting the server to sanitize the values

File: src/models/room-state.js (hunk 1 above)
Comment (t3chguy) [reply src/models/room-state.js]: I got impatient and switched it and the others in this file over to a utils method `isNumber` I added which does pretty much the above

File: src/utils.js [hunk 2]
```diff
@@ -652,3 +652,13 @@ module.exports.inherits = function(ctor, superCtor) {
         },
     });
 };
+
+/**
+ * Returns whether the given value is a finite number without type-coercion
+ *
+ * @param {*} value the value to test
+ * @return {boolean} whether ot not value is a finite number without type-coercion
```
Comment (lukebarnard1) [inline src/utils.js]: "ot"

**Comment by lukebarnard1** [comment]
> I guess here we're trusting the server to sanitize the values

I don't think we are doing that here.

>but did a not NaN previously

I can't see evidence of us doing that before. `isFinite(NaN)` returns false so I think we're fine here.

**Comment by t3chguy** [comment]
> but did a not NaN previously

as in a local change before this commit I made, then went for `!== undefined` to match the rest of the class, then refactored the class

**Pull request merged**